src/core_plugins/elasticsearch/lib/kibana_index_patches/patch_missing_title_keyword_fields.js

```diff
@@ -0,0 +1,68 @@
+import { getRootProperties } from '../../../../server/mappings';
+import { get } from 'lodash';
+import { KibanaIndexPatch } from './kibana_index_patch';
+
+const propertiesWithTitles = [
+  'index-pattern',
+  'dashboard',
+  'visualization',
+  'search',
+];
+
+export class PatchMissingTitleKeywordFields extends KibanaIndexPatch {
```

Comment:
Personally, I think that subclassing is kind of a confusing way to configure something. It also makes it more difficult for plugins to add their own configurations if we decide to expose this to plugins down the road.

What if we just exported configurations that received a context of some sort as an argument? `patchKibanaIndex` could create instances of `KibanaIndexPatch` with these configurations if desired, maybe validating the options and such.

```js
export const missingTitleKeywordFieldsPath = {
  async getUpdatedPatchMappings({ currentMappingsDsl }) {
    const properties = getRootProperties(currentMappingsDsl);
    const mappings = {};

    for (const property of propertiesWithTitles) {
      const hasKeyword = !!get(properties, `${property}.properties.title.fields.keyword`);
      if (hasKeyword) {
        continue;
      }

      const titleMapping = get(properties, `${property}.properties.title`);
      mappings[property] = {
        properties: {
          title: {
            ...titleMapping,
            fields: {
              keyword: {
                type: 'keyword',
              }
            }
          }
        }
      };
    }

    // Make sure we return a falsy value
    if (!Object.keys(mappings).length) {
      return;
    }

    return mappings;
  },
  
  async applyChanges(context) {
    const {
      log,
      callCluster,
      patchedMappings,
      indexName,
      rootEsType,
    } = context;
    
    const properties = Object.keys(patchedMappings);
    const types = properties.map(type => ({ match: { type } }));
 
    log(['info', 'elasticsearch'], {
      tmpl: `Updating by query for Saved Object types "<%= names.join('", "') %>"`,
      names: properties,
    });
 
    await callCluster('updateByQuery', {
      conflicts: 'proceed',
      index: indexName,
      type: rootEsType,
      body: {
        query: {
          bool: {
            should: types,
          },
        },
      },
    });
  }
})
```

Reply:
Cool, thanks for the suggestion. I've been in the FieldFormat stuff recently and don't want to walk that path if I can avoid it. I'll work on this